public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx

```diff
@@ -13,6 +20,37 @@ type Props = {
 
 export function PluginDetailsRightPanel(props: Props): React.ReactElement | null {
   const { info, plugin } = props;
+  const [defaultBranch, setDefaultBranch] = useState<string | null>(null);
```

Comment:
Echoing @tolzhabayev's idea here, all this logic should move to gcom and we should directly fetch the links, let the gcom API figure all this and grafana simply show the links. 

Some additions:

*In github you can simply use `HEAD` instead of trying to fetch the default branch from it. e.g. https://github.com/grafana/grafana/blob/HEAD/LICENSE 
* we should add support for bitbucket (some repositories there too)